frontend/readhistory.lua

```diff
@@ -23,30 +23,14 @@ end
 
 local function buildEntry(input_time, input_file)
     local file_path = realpath(input_file) or input_file -- keep orig file path of deleted files
+    local is_file_deleted = lfs.attributes(file_path, "mode") ~= "file"
     return {
         time = input_time,
         file = file_path,
         text = input_file:gsub(".*/", ""),
-        dim = lfs.attributes(file_path, "mode") ~= "file", -- "dim", as expected by Menu
-        mandatory_func = function() -- Show the last read time
-            local readerui_instance = require("apps/reader/readerui"):_getRunningInstance()
-            local currently_opened_file = readerui_instance and readerui_instance.document and readerui_instance.document.file
-            local last_read_ts
-            if file_path == currently_opened_file then
-                -- Don't use the sidecar file date which is updated regularly while
-                -- reading: keep showing the opening time for the current document.
-                last_read_ts = input_time
-            else
-                -- For past documents, the last save time of the settings is better
-                -- as last read time than input_time (its last opening time, that
-                -- we fallback to if no sidecar file)
-                last_read_ts = DocSettings:getLastSaveTime(file_path) or input_time
-            end
-            return datetime.secondsToDateTime(last_read_ts, G_reader_settings:isTrue("twelve_hour_clock"))
-        end,
-        select_enabled_func = function()
-            return lfs.attributes(file_path, "mode") == "file"
-        end,
+        dim = is_file_deleted,
+        mandatory = datetime.secondsToDateTime(input_time),
+        select_enabled = not is_file_deleted,
```

Comment:
No need for `_func` ? If we delete a file from history (with the option "removed deleted file from history" disabled), are we sure we go thru this buildEntry() to get updated info ?
edit: OK, I see below that you update this field when deleting. Keeping that comment just for you to answer "all is good and thought of" :)